### PR TITLE
[WIP] Adding support of meilisearch 

### PR DESCRIPTION
**Pull request by kcubeterm**
meili.js: initialise
crawler.js: some modifications, now crawler will split files in multiple chunks of 80MB each

### CODE REVIEW

File: crawler/crawler.js [hunk 1]
```diff
@@ -128,9 +144,9 @@ function universalFileSwitcher(filePath) {
             docHandler(filePath);
             break;
         case '.docx':
-        case '.odt':
-        case '.odp':
-        case '.ods':
+        // case '.odt':
```
[the remark this answers is not in the document]
Comment (cauerego) [reply crawler/crawler.js:147]: better to insert this comment (about why those lines were commented out) in the file itself. versioning conversations, specially these exclusive to github, can get lost way more easily than the file.

**Review by cauerego — COMMENTED** [review]
i love the comment description there.

file name such as this, for a class, usually should take the same name as the class, to prevent confusion.

and the comment about the last method seem to apply for the whole class, doesn't it? if so, should be moved to the top.

beautiful work!

**Comment by kcubeterm** [comment]
> and the comment about the last method seem to apply for the whole class, doesn't it? 

Yes, I will move that on top to avoid confusion.

**Comment by cauerego** [comment]
my bad, didn't mean to mark it as "ready to review", thought i would just open a review process. 🤣

**Comment by kcubeterm** [comment]
while adding meilisearch support. it is obvious that typesense has break. that is why this wrapper would be compatible with both search engine.

**Comment by cauerego** [comment]
well, the production "demo" ( achoz.ahoxus.org ) doesn't work right now.

**Comment by kcubeterm** [comment]
> well, the production "demo" ( achoz.ahoxus.org ) doesn't work right now.

even typesense is buggy too. let me check the log.

**Comment by kcubeterm** [comment]
or more better, just focus on meilisearch for now until achoz is in beta stage. once it become more stable. will think of typesense and other searchengine.

**Comment by cauerego** [comment]
i think it's definitely better to focus on 1 engine only, and get it to work.

**Comment by kcubeterm** [comment]
this PR is almost finished. Just need to test with latest meilisearch and bit of fixes if needed while testing.

**Pull request closed without merging**